### PR TITLE
chore(appveyor): build x86 on real x86 machines

### PR DESCRIPTION
**Pull request by jviotti**
We do this by using Appveyor's `platform` configuration variable instead
of always running on x64 and cross-compiling to x86.

Change-Type: patch
Signed-off-by: Juan Cruz Viotti <jv@jviotti.com>

### CODE REVIEW

File: scripts/ci/build-installers.sh [hunk 1]
```diff
@@ -46,10 +46,11 @@ fi
 
 if [ "$ARGV_OPERATING_SYSTEM" == "linux" ]; then
   ./scripts/build/docker/run-command.sh \
-    -r "$TARGET_ARCH" \
+    -r "$ARGV_ARCHITECTURE" \
```
Comment (lurch) [inline scripts/ci/build-installers.sh]: Haha, I'd also spotted this discrepancy and had this change in a WIP branch that I haven't pushed yet ;-)

**Comment by lurch** [comment]
LGTM, if all the tests pass. Does the fact that the changes have been made to `appveyor.yml` mean that we won't know if this PR is working correctly until after it's been merged to `master`?

**Comment by lurch** [comment]
That's weird - https://ci.appveyor.com/project/resin-io/etcher/build/1.0.6751/job/405y2uk32ykdmjha says `Platform: x86` at the top, but then in the build-log it displays:
```
Application version : 1.2.1+c4f8c27
Release type        : snapshot
Platform            : win32
Host arch           : x64
Target arch         : x86
```

**Comment by jviotti** [comment]
> LGTM, if all the tests pass. Does the fact that the changes have been made to appveyor.yml mean that we won't know if this PR is working correctly until after it's been merged to master?

The builds are building the new Appveyor file.

> That's weird - https://ci.appveyor.com/project/resin-io/etcher/build/1.0.6751/job/405y2uk32ykdmjha says Platform: x86 at the top, but then in the build-log it displays:

Hm, maybe we're detecting the arch incorrectly on Windows? I'll take a look.

**Comment by lurch** [comment]
> The builds are building the new Appveyor file.

Hmmm, I thought there was something that meant that changes to `appveyor.yml` and `.travis.yml` didn't have any affect until after they'd been merged to master, to prevent a 'malicious' PR from a 3rd-party user from doing 'bad' things by editing these files? Or perhaps I'm getting confused with something else...?

**Comment by lurch** [comment]
I've just noticed that even on the merge-to-master build, the 'x86' Appveyor logfile says:
```
Chocolatey v0.10.7
Installing the following packages:
jq
By installing you accept licenses for the packages.
Progress: Downloading jq 1.5... 100%
jq v1.5 [Approved]
jq package files install completed. Performing other installation steps.
WARNING: Url has SSL/TLS available, switching to HTTPS for download
Downloading jq 64 bit
  from 'https://github.com/stedolan/jq/releases/download/jq-1.5/jq-win64.exe'
Progress: 100% - Completed download of C:\ProgramData\chocolatey\lib\jq\tools\jq.exe (2.22 MB).
Download of jq.exe (2.22 MB) completed.
```
so it looks like it really **is** still running on a x64 host (since IIRC 64-bit code can't run on a 32-bit host), so I'm not sure what the `Platform` option is actually doing? \*shrug\*

**Comment by jviotti** [comment]
> Hmmm, I thought there was something that meant that changes to appveyor.yml and .travis.yml didn't have any affect until after they'd been merged to master, to prevent a 'malicious' PR from a 3rd-party user from doing 'bad' things by editing these files? Or perhaps I'm getting confused with something else...?

They take effect, but they don't have access to any secure environment variables configured on the CI services.

> so it looks like it really is still running on a x64 host (since IIRC 64-bit code can't run on a 32-bit host), so I'm not sure what the Platform option is actually doing? *shrug*

Ah, strange. Maybe they just configure all the tooling (Visual Studio, WinDDK, etc) for 32-bit?

**Pull request merged**